modules/tensor_mechanics/src/materials/lagrangian/ComputeHypoelasticStVenantKirchhoffStress.C

```diff
@@ -0,0 +1,78 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ComputeHypoelasticStVenantKirchhoffStress.h"
+
+registerMooseObject("TensorMechanicsApp", ComputeHypoelasticStVenantKirchhoffStress);
+
+InputParameters
+ComputeHypoelasticStVenantKirchhoffStress::validParams()
+{
+  InputParameters params = ComputeLagrangianObjectiveStress::validParams();
+  params.addClassDescription(
+      "Calculate a small strain elastic stress that is equivalent to the hyperelastic St. "
+      "Venant-Kirchhoff model if integrated using the Truesdell rate.");
+
+  params.addParam<MaterialPropertyName>(
+      "elasticity_tensor", "elasticity_tensor", "The name of the elasticity tensor.");
+
+  return params;
+}
+
+ComputeHypoelasticStVenantKirchhoffStress::ComputeHypoelasticStVenantKirchhoffStress(
+    const InputParameters & parameters)
+  : ComputeLagrangianObjectiveStress(parameters),
+    _elasticity_tensor(getMaterialProperty<RankFourTensor>(
+        getParam<MaterialPropertyName>(_base_name + "elasticity_tensor"))),
+    _def_grad(getMaterialProperty<RankTwoTensor>(_base_name + "deformation_gradient"))
+{
+}
+
+void
+ComputeHypoelasticStVenantKirchhoffStress::computeQpSmallStress()
+{
+  // The increment in the spatial velocity gradient
+  RankTwoTensor dL = _strain_increment[_qp] + _vorticity_increment[_qp];
+
+  // If small kinematics, it falls back to the grade-zero hypoelasticity
+  if (!_large_kinematics)
+  {
+    _small_stress[_qp] = _small_stress_old[_qp] + _elasticity_tensor[_qp] * dL;
+    _small_jacobian[_qp] = _elasticity_tensor[_qp];
+    return;
+  }
+
+  // Large kinematics:
+  // First push forward the elasticity tensor
+  const RankTwoTensor F = _def_grad[_qp];
+  const RankTwoTensor Ft = F.transpose();
+  const Real J = F.det();
+  const RankFourTensor FF = F.mixedProductIkJl(F);
+  const RankFourTensor FtFt = Ft.mixedProductIkJl(Ft);
+  const RankFourTensor C0 = _elasticity_tensor[_qp];
+  const RankFourTensor C = FF * C0 * FtFt / J;
+
+  // Update the small stress
+  const RankTwoTensor dS = C * dL;
+  _small_stress[_qp] = _small_stress_old[_qp] + dS;
+
+  // Compute the small Jacobian
+  if (_fe_problem.currentlyComputingJacobian())
```

Comment:
Uhh, do we want to start introducing this now or do it all at once later on?

Reply:
Let me do the right thing here. I'll correct everything else in a separate PR.